File: Clases 2018/Clases-C#/ClasePractica12_2018 (transacciones con data manager)/C#_TPS_BugTracker/TPS_BugTracker/BusinessLayer/Services/BugService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;
using TPS_BugTracker.DataLayer.DAOs;

namespace TPS_BugTracker.BusinessLayer.Services
{
    public class BugService
    {
        private BugDao oBugDao;
        private HistorialBugDao oHistorialBugDao;
        public BugService()
        {
            oBugDao = new BugDao();
            oHistorialBugDao = new HistorialBugDao();
        }
        public IList<Bug> consultarBugsConFiltros(List<object> @params)
        {
            return oBugDao.getBugByFilters(@params);
        }

        public Bug consultarBugsPorId(int id)
        {
            return oBugDao.getBugById(id);
        }

        public bool actualizarBug(Bug oBug)
        {
            return oBugDao.update(oBug);
        }
    }
}

File: PAV 1-TPI - CASOS DE USO version 1/Resueltos/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/BusinessLayer/Services/BugService.vb
Public Class BugService
    Private oBugDao As BugDao
    Public Sub New()
        oBugDao = New BugDao
    End Sub

    Public Function create(ByVal oBug As Bug) As Boolean
        Return oBugDao.create(oBug)
    End Function

    Public Function consultarBugsConFiltros(ByVal params As Object()) As List(Of Bug)
        Return oBugDao.getBugByFilters(params)
    End Function

    Public Function consultarBugsPorId(ByVal id As Integer) As Bug
        Return oBugDao.getBugById(id)
    End Function

    Public Function actualizarBugConHistorial(ByVal lst As List(Of Bug)) As Boolean
        Return oBugDao.updateNewHistorial(lst)
    End Function

End Class
